### DISCUSSION TITLE
OpenFAST version

### DISCUSSION THREAD

**Discussion by Javier-TRI** [IEAWindTask52/LidarAssistedControl#15] [Q&A] [answered]
Hey everyone!

Just wondering, does anyone remember which version of OpenFAST we’re on? My memory’s drawing a blank…

We’re in the middle of the work, and we’re trying to make sure our stuff will play nice with the OpenFAST version for the Summer Games.

Appreciate any help!

Javi

**Comment by josujugo** [comment, marked as answer]
Hi, Javi:

I think that the version is OpenFAST3.0_Lidarsim. A windows  executable is in the github space.

josu

**Comment by Javier-TRI** [comment]
Hi everyone,

Josu you’re correct. Thank you for your response... I didn't check that OpenFAST version... 👍🏼 

We are currently utilizing the same version of OpenFAST, albeit it’s not the one available on GitHub. Upon reviewing the source code, I’ve noticed that the primary distinction is the inclusion of the LIDAR module. Consequently, I plan to adjust the provided source code to accommodate our methodology.

I’m curious about the application of this new module. Is it employed to simulate the LIDAR, right? If we were to incorporate the measured wind speed into our control strategy, would we obtain this wind speed data from this module? Alternatively, is the wind data processed in a different module?

In the LIDAR module’s input file, I noticed a variable named ‘V_LOS’. Could anyone confirm if this variable represents the wind speed measured by the LIDAR?

Thank you in advance.

Best,

Javi

**Comment by DavidSchlipf** [comment]
Hi Javi, thanks for using the Forum!
The OpenFAST version includes a detailed lidar simulator based on Feng's work: 
https://iopscience.iop.org/article/10.1088/1742-6596/2265/4/042030 
The repository for this version can be found at:
https://github.com/DavidSchlipf/OpenFAST3.0_LidSim/tree/OpenFAST4.0_merge
We are working with NREL to see, how this can be implemented as a totally independent OpenFAST module. 
And yes, the signal V_LOS is the line-of-sight wind speed depeding on the lidar system specified in the lidar input file. We added one a typical circular CW system and for a typical 4-Beam lidar system. You need to process it with a lidar-data-processing to get e.g.  rotor-effective wind speed (REWS), e.g. the LDP_v1.dll. 
Please let me know, if you need more information! And great that you are working on it, looking forward to the results! Are you working on the 30s sprint, 18 m/s hurdles or the DLC 1.2 marathon?
Best regards
David

**Reply by Javier-TRI** [reply]
Hi David,

Thanks for getting back to me so quickly! 

I'm going to dive into that publication and get a better handle on what's happening with the LIDAR simulator. I think I need to dig a bit deeper into this since our current approach is Python-based and we'll need to find a way to integrate the LDP DLL for the V_LOS processing.

Fingers crossed, if all goes according to plan, we'll have some results to show off at the sprint and the marathon. We've got a student on board who's eager to compete at the student level, and Josu and I are rolling up our sleeves for the research competition.

Best,

Javi

**Reply by DavidSchlipf** [reply]
Hi Javi,
You could check the Fortran Source Code of the LDP DLL in the repository and implement it directly!
And nice you plan to contribute to both!
David

**Reply by Javier-TRI** [reply]
David,
I’ve been examining the Fortran Source Code of the LDP and so far, everything seems to be clear. However, we’ve run into some obstacles during the OpenFAST compilation process for our working approach.
I’m going to investigate these issues further. If I need any assistance, I’ll reach out on the forum.
Additionally, if you happen to compile ACADOS and integrate the predictive controller into the wrapper workflow, could you please let me know? I’m interested in implementing the same at some point.
Best, 
Javi

**Reply by josujugo** [reply]
Hi, All:

I am checking all the code  and I am wondering if someone is using OpenFAST3.0_LidSim integrated with Simulink, that is, with the compiled library, not the EXE file.

Thanks

josu

**Reply by DavidSchlipf** [reply]
Hi Javi and Josu,
Maybe Feng can comment on the Simulink version. And regarding the compilation of Acados to a dll, hopefully Simon or Chris can give you some hints.
David

**Reply by fengguoFUAS** [reply]
Hi All

I just compiled a Simulink version of the OpenFAST to be used in this game. I also added an example code showing how to run it with IEA15MW_01.  You should be able to access it soon once my push is accepted.

BR
Feng Guo

**Reply by DavidSchlipf** [reply]
Thanks Feng, it is already on the main!

**Comment by josujugo** [comment]
Hi, again:

@fengguoFUAS, I am wondering if it is possible for you to compile the openfastlibrary.dll. I am trying to access the library from python.


If this is possible, I would like to check the sources, to try some small changes in the code. Is this posible?, That is, is the code used for compiling the Simulink library accesible?

Best regards

josu

**Reply by fengguoFUAS** [reply]
Dear Josu,

The source code to compile the simulink library and executable is available here: https://github.com/DavidSchlipf/OpenFAST3.0_LidSim/tree/OpenFAST4.0_merge

I compiled the simulink S-function using VS 2019 and Inter visual fortran compiler 2021.

Unfortunately, I have not worked with compiling it as a dll that could be directly called by python. Maybe you can check the OpenFAST forum to see if other people have done it and you might get some clues from there.

BR
Feng Guo

**Comment by josujugo** [comment]
Thank you Feng. I will try to compile again from this code.

Best regards

josu
